distributedcombigrid/src/sgpp/distributedcombigrid/fullgrid/DistributedFullGrid.hpp

```diff
@@ -201,12 +201,103 @@ class DistributedFullGrid {
     MPI_Comm_free(&communicator_);
   }
 
+  FG_ELEMENT evalLocalIndexOn(const IndexVector& localIndex, const std::vector<real>& coords) const {
+    auto firstIndex = IndexVector(dim_, 0);
+    auto lastIndex = this->getLastGlobalIndex() - this->getFirstGlobalIndex();
+    // if this local index is out of bounds, return 0. (will be contributed by other partial dfg)
+    if (! (localIndex >= firstIndex && localIndex <= lastIndex) ) {
+      // std::cout << "out of bounds" << localIndex << firstIndex << lastIndex << std::endl;
+      return 0.;
+    }
+
+    // get coords corresponding to localIndex
+    auto localLinearIndex = getLocalLinearIndex(localIndex);
+    std::vector<real> pointCoords (this->getDimension());
+    getCoordsLocal(localLinearIndex, pointCoords);
+
+    // get product of 1D hat functions on coords
+    auto h = getGridSpacing();
+    real phi_c = 1.;
```

Comment:
Maybe add description what phi_c is (value of basis function at coords). This would increase readability.

Reply:
good idea. will add that.